stakedb/stakedb.go

```diff
@@ -454,6 +529,20 @@ func (db *StakeDatabase) PoolSize() int {
 	return db.BestNode.PoolSize()
 }
 
+// PoolAtHeight gets the entire list of live tickets at the given chain height.
+func (db *StakeDatabase) PoolAtHeight(height int64) ([]chainhash.Hash, error) {
```

Comment:
What happens here during a fork?

Reply:
The stake database always follows what the node thinks is the main chain.  On a reorg, stakedb.go trims back the ticket pool db and appends the new main chain.  The node doesn't convey any information about forks that don't cause a reorg, so we don't have a way to look up those blocks even if we wanted to.